server/src/test/java/org/elasticsearch/health/node/FetchHealthInfoCacheActionTests.java

```diff
@@ -0,0 +1,163 @@
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import static org.elasticsearch.test.ClusterServiceUtils.createClusterService;
+import static org.elasticsearch.test.ClusterServiceUtils.setState;
+import static org.hamcrest.Matchers.equalTo;
+
+public class FetchHealthInfoCacheActionTests extends ESTestCase {
+    private static ThreadPool threadPool;
+
+    private ClusterService clusterService;
+    private TransportService transportService;
+    private DiscoveryNode localNode;
+    private DiscoveryNode[] allNodes;
+
+    @BeforeClass
+    public static void beforeClass() {
+        threadPool = new TestThreadPool("FetchHealthInfoCacheAction");
+    }
+
+    @Before
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        CapturingTransport transport = new CapturingTransport();
+        clusterService = createClusterService(threadPool);
+        transportService = transport.createTransportService(
+            clusterService.getSettings(),
+            threadPool,
+            TransportService.NOOP_TRANSPORT_INTERCEPTOR,
+            x -> clusterService.localNode(),
+            null,
+            Collections.emptySet()
+        );
+        transportService.start();
+        transportService.acceptIncomingRequests();
+        int totalNodes = randomIntBetween(1, 200);
+        allNodes = new DiscoveryNode[totalNodes];
+        localNode = new DiscoveryNode(
+            "local_node",
+            buildNewFakeTransportAddress(),
+            Collections.emptyMap(),
+            Set.of(DiscoveryNodeRole.MASTER_ROLE, DiscoveryNodeRole.DATA_ROLE),
+            Version.CURRENT
+        );
+        allNodes[0] = localNode;
+        for (int i = 0; i < totalNodes - 1; i++) {
+            DiscoveryNode remoteNode = new DiscoveryNode(
+                "remote_node" + i,
+                buildNewFakeTransportAddress(),
+                Collections.emptyMap(),
+                Set.of(DiscoveryNodeRole.MASTER_ROLE, DiscoveryNodeRole.DATA_ROLE),
+                Version.CURRENT
+            );
+            allNodes[i + 1] = remoteNode;
+        }
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        super.tearDown();
+        clusterService.close();
+        transportService.close();
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        ThreadPool.terminate(threadPool, 30, TimeUnit.SECONDS);
+        threadPool = null;
+    }
+
+    public void testAction() throws ExecutionException, InterruptedException {
+        FetchHealthInfoCacheAction.Request request = new FetchHealthInfoCacheAction.Request();
+        PlainActionFuture<FetchHealthInfoCacheAction.Response> listener = new PlainActionFuture<>();
+        setState(clusterService, ClusterStateCreationUtils.state(localNode, localNode, localNode, allNodes));
```

Comment:
Hm, I was wondering, this test case is health node is the local node. Should we test retrieving the health info from a remote health node? What do you think?

Reply:
I don't think the location of the health node matters for this code that's being tested does it? It definitely matters for TransportHealthNodeAction, but this code can safely assume that TransportHealthNodeAction works and has been tested right? There's no branching in FetchHealthInfoCacheAction itself that's dependent on which node is the health node.

Reply:
Hm, I see. Okay.